src/main/java/com/sem/btrouble/model/Model.java

```diff
@@ -65,26 +68,16 @@ public static Timers getTimers() {
     }
 
     /**
-     * Adds a room to the Model's room collection.
-     *
-     * @param room
-     *            should be the Room which is added to the Model's room
-     *            collection
-     */
-    public static void addRoom(Room room) {
-        rooms.add(room);
-    }
-
-    /**
      * Get the next room.
      * 
      * @return returns the next room
      */
     public static Room getNextRoom() {
         currentLevel++;
-        if (rooms.hasNext()) {
-            return rooms.next();
+        if (dataLoader.hasRoom(currentLevel)) {
```

Comment:
Could also be done with if(dataLoader.hasNext()), where the currentLevel is recorded in the DataLoader Object. The current solution is fine though.